### PR TITLE
Update Rust and crates

### PR DESCRIPTION
**Pull request by BlackDex**
- Updated Rust to v1.77.0
- Updated several crates
  The `reqwest` update included `trust-dns` > `hickory-dns` changes.
  Also, `reqwest` v0.12 is not working correctly for us, that is something to investigate.
- Fixed a new clippy warning

### CODE REVIEW

**Review by dani-garcia — APPROVED** [review]
LGTM, what's the issue with reqwest 0.12?

File: Cargo.toml [hunk 1]
```diff
@@ -123,7 +123,7 @@ email_address = "0.2.4"
 handlebars = { version = "5.1.0", features = ["dir_source"] }
 
 # HTTP client (Used for favicons, version check, DUO and HIBP API)
-reqwest = { version = "0.11.26", features = ["stream", "json", "gzip", "brotli", "socks", "cookies", "trust-dns", "native-tls-alpn"] }
+reqwest = { version = "0.11.27", features = ["default-tls", "native-tls-alpn", "stream", "json", "gzip", "brotli", "socks", "cookies", "hickory-dns"], default-features = false}
```
Comment (dani-garcia) [inline Cargo.toml:126]: I don't think default-tls needs to be specified, as it already is enabled by native-tls-alpn, but no big deal either way:

https://github.com/seanmonstar/reqwest/blob/cf69fd4bfe22855d576497eb94e9eb549e742475/Cargo.toml#L37-L39

Comment (BlackDex) [reply Cargo.toml:126]: Ah, yes, that is true indeed.

**Comment by BlackDex** [comment]
> LGTM, what's the issue with reqwest 0.12?

It doesn't seem to download almost any icon.
I suspect http2 maybe, or some other changes.

Also, other crates we use still use reqwest 0.11.x, so we would introduce duplicate dependencies/crates.

**Pull request merged**